### ISSUE TITLE
InternalIcon fails in TreeView in runtime due to no methods

### ISSUE THREAD

**Issue by jgfoster** [dolphinsmalltalk/Dolphin#788]
**Describe the bug**
InternalIcon instances are no longer working in a runtime image. It appears that all their methods are stripped, but I can't tell where:

`InternalIcon selectors size "16 in development"`

The image stripping log shows that `#imageIndex` is recognized as a sent message, my instances are present, but the class has no methods:

```
			<Class name="InternalIcon" count="8" memoryUsage="192" unbound="true">
				<Methods count="0">
				</Methods>
				<ClassMethods count="0">
				</ClassMethods>
				<References>
					<Reference identityHash="421003264">an InternalIcon</Reference>
```

**Please complete the following information):**
 - OS: [e.g. Windows 10]

**Additional context**
As a work-around I was able to create a subclass of InternalIcon and add the method to it.

```
{1E5A1008: cf 1E5A0FE9, sp 1E5A101C, bp 1E5A1004, ip 4, InternalIcon(Object)>>doesNotUnderstand:}
	receiver: a InternalIcon
	arg[0]: Message selector: imageIndex arguments: 01D82CE0

{1E5A0FE8: cf 1E5A0FCD, sp 1E5A0FFC, bp 1E5A0FE8, ip 3, Project(Object)>>iconImageIndex}
	receiver: a Project

{1E5A0FCC: cf 1E5A0FAD, sp 1E5A0FE0, bp 1E5A0FC8, ip 4, IconicListAbstract class>>value:}
	receiver: IconicListAbstract
	arg[0]: a Project

{1E5A0FAC: cf 1E5A0F79, sp 1E5A0FC0, bp 1E5A0F94, ip 39, PCTreeView(TreeView)>>onDisplayDetailsRequired:}
	receiver: a PCTreeView
	arg[0]: a TVITEMW
	stack temp[0]: a Project
	stack temp[1]: 67
	stack temp[2]: false
	stack temp[3]: IconicListAbstract
	stack temp[4]: nil

{1E5A0F78: cf 1E5A0F59, sp 1E5A0F8C, bp 1E5A0F74, ip 5, PCTreeView(IconicListAbstract)>>nmGetDispInfoW:}
	receiver: a PCTreeView
	arg[0]: a ExternalAddress

{1E5A0F58: cf 1E5A0F35, sp 1E5A0F6C, bp 1E5A0F50, ip 27, PCTreeView(TreeView)>>nmNotify:}
	receiver: a PCTreeView
	arg[0]: a ExternalAddress
	stack temp[0]: nmGetDispInfoW:
```

**Comment by blairmcg** [comment]
The key bit of information in the class entry from the log in this case is the `unbound="true"`, which means that the class has been uninstalled. It's only present because of the instances. Instances alone are not sufficient to retain something. There would need to be a reference to InternalIcon from some method that is reachable along a code path from one of the "root" packages of the application. I've probably removed a code path that was keeping this there before.  If you'd like to share the full deployment log, I could probably tell you why.

You might wish to do your own fishing, however, in which case I'd recommend starting with the set of required packages listed at the top of the log. If the internal icons package is not among those, then shortly below you should see the notification relating to it being removed. The set of required package is the transitive closure of package dependencies, so you should be able to determine whether the package is somewhere among the pre-requisites of the deployed package by expanding the pre-reqs in the package browser. In order for a package to be a pre-requisite, it either has to be an explicitly listed manual pre-requisite (rarely needed), or there would need to be a reference path to one of the classes or globals.

If the package is among the pre-requisites, then the class must have been removed later as part of the cycle of progressively trimming out unsent methods, unreferenced classes (repeat). This will show up as a class removal entry in the log.

I often find it useful to open an Environment Browser on the log to see what is actually in the deployed application in terms of classes and methods. This can be done either by using the Deployment/Browse Executable Manifest command from the context menu of a package in the package browser, or programmatically by evaluating an expression like this:
```
Smalltalk developmentSystem browseDeploymentLog: 'c:\wherever\whatever.xml'
```
From this you can determine why something is in the application, by looking at references to some particular class or method, or perhaps why it is not by determining that a key referencing method has been removed.

**Comment by blairmcg** [comment]
I think this will not repro if you have a code path to InternalIcon. As an example I added an example script to the Plimbole package comment to deploy it, and it uses InternalIcons. It works fine because these are returned by methods. I think it would also be fine if the InternalIcon were stored in a view resource. 
It would not work if the InternalIcons have been loaded as instances into class variables (say), and the class initialization method has been removed (which will be the case by default, unless you add it to the 'must not strip' category - see `BasicImageStripper>>removeNotRequiredMethods`). The fact that some instances may exist will not prevent the standard deployment process from removing classes.
In other words, this would appear to be behaving as expected.

**Comment by jgfoster** [comment]
I've added an explicit reference to InternalIcon (sending #'yourself') in my code and this seems to have solved it. Thanks for the research and explanation.
